### PR TITLE
fix a few gramma errors in API documentation

### PR DESCRIPTION
**Pull request by contrun**
I found multiple typos in `API.md`. A few sentences are quite elusive to me. I changed them liberally. Please correct me if there are any distortions. I also found one comment (annotated with a TODO) repeated in different contexts,.

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -698,10 +699,10 @@ Header flags of xqc_http3_nv_flag_t with OR operator, see xqc_http3_nv_flag_t.
 Array of headers.
 
 #### count
-header count of headers.
+Nuber of headers.
```
Comment (Kulsk) [inline docs/API.md:702]: number

File: docs/API.md [hunk 2]
```diff
@@ -608,6 +608,7 @@ Transfer dcid to human-readable string.
 The read notify flag of _xqc_h3_request_read_notify_pt_ callback function.
 
 #### XQC_REQ_NOTIFY_READ_NULL
+TODO: This is the same as the below field.
```
Comment (Kulsk) [inline docs/API.md:611]: NULL flag, stands for nothing to be read. please help to specify this

**Comment by Kulsk** [comment]
> I found multiple typos in `API.md`. A few sentences are quite elusive to me. I changed them liberally. Please correct me if there are any distortions. I also found one comment (annotated with a TODO) repeated in different contexts,.

Great thank! you  did a wonderful job that really saved us! It looks much better and easier to understand now.

**Review by Kulsk — APPROVED** [review]

**Comment by RuiqiZhou** [comment]
Thanks for your contribution, but would you mind changing the commit messages to 
`[=] fix a few gramma errors in API documentation`
as mentioned in https://github.com/alibaba/xquic/blob/main/CONTRIBUTING.md#commit-messages

**Review by RuiqiZhou — APPROVED** [review]
LGTM

**Pull request merged**